src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -310,10 +321,12 @@ std::string tenantFromDb(std::string& database)
 {
   std::string r;
   std::string prefix  = dbPrefix + "-";
+
   if (strncmp(prefix.c_str(), database.c_str(), strlen(prefix.c_str())) == 0)
   {
     char tenant[MAX_SERVICE_NAME_LEN];
```

Comment:
Not sure... maybe you could have a look to the documentation and calculate the maximum size taking into account all the rules we have (number of levels, max lengh in each level, etc.).